### PR TITLE
OpenAPI work for MDS

### LINKED ISSUE #281: Use OpenAPI to define MDS API request shapes and query parameters
The existing documentation of query parameters and headers is in narrative format in the Provider and Agency README files. This is useful to give an overview of what's allowed, but because it is not programmatically enforced and left up to interpretation, it's common to see divergences from the spec. [1]

I'd like to propose adding API endpoint parameter definitions to Provider (and maybe Agency?) endpoints using a format designed to support automatic validation. 

Implementers of MDS endpoints and software that interacts with them would be able to use these definitions to validate the shape of query parameters and headers for incoming requests. Additionally, a rigorous definition of the request shape would let MDS evolve to support more advanced request formats in a safe and transparent way.

A framework frequently used for this kind of schema is [OpenAPI v3](https://swagger.io/docs/specification/describing-parameters/), an evolution of Swagger API definition formats. It has a lot of OSS and third party tools for API documentation and validation that may come in useful, and is reasonably well documented. Based on prior experience, I'd recommend using this format. [2]

---

[1] A similar thing to this already exists for validating the shape of responses via JSON Schema, and has been extremely useful for deciding if an API response conforms to the specification. However, JSON Schema is a toolkit for asserting the shape of JSON objects and  does not really support defining the shape of request parameters and headers. 

[2] I also considered suggesting [JSON Hyper Schema](https://json-schema.org/latest/json-schema-hypermedia.html) ([narrative intro](https://blog.apisyouwonthate.com/getting-started-with-json-hyper-schema-184775b91f)), an extension to JSON Schema that builds a vocabulary for defining URI shapes. However, it is very young and does not have a really mature ecosystem of tools around it yet.

### LINKED ISSUE #792: OpenAPI: Provide json with vehicle state machine
Hi, there. I'm working on writing a validator to make sure a vehicle goes from state to state through valid events and I was wondering if it would be helpful to include a `.json` file in the repo with an object that could be used to validate transitions. Something like this:

```json
{
  "previous_state": {
    "new_state": ["valid_event_1", "valid_event_2", ...],
    ...
  },
  "non_operational": {
    "removed": ["maintenance_pick_up", "rebalance_pick_up", ...],
    "available": ["battery_charged", "maintenance", ...],
    "unknown": ["comms_lost", "missing", "unspecified"],
  },
}
```
I think it would help people building validators to have a file like this as a single source of truth in the repo.

### PR DESCRIPTION
**Pull request by schnuerle**
## Explain pull request

Removing JSON Schema from MDS repository and instead moving to OpenAPI in a new external repository, [mds-openapi](https://github.com/openmobilityfoundation/mds-openapi).

## Is this a breaking change

* Yes, breaking

## Impacted Spec

Which spec(s) will this pull request impact?

* All

## Additional context

To see the most recent development changes to the JSON Schema while the community was working on MDS 2.0, see the [snapshot-jsonschema](https://github.com/openmobilityfoundation/mobility-data-specification/tree/snapshot-jsonschema) feature branch.

### CODE REVIEW

**Comment by schnuerle** [comment]
Note for this work, references to JSON Schema within each API page will need to be updated with direct links, including general schema references and per endpoint.

**Comment by schnuerle** [comment]
Note to remove in page references to JSON Schema and update to OpenAPI, fix vehicle data type like on Provider page, add some mention of validation on the MDS home page in a new section.

File: README.md [hunk 1]
```diff
@@ -123,6 +124,12 @@ The MDS specification is versioned using Git tags and [semantic versioning](http
 * [MDS Releases](https://github.com/openmobilityfoundation/governance/wiki/Releases) - current/recommended versions, timeline
 * [Release Guidelines](https://github.com/openmobilityfoundation/governance/blob/main/technical/ReleaseGuidelines.md)
 
+## Data Validation
+
+To help with MDS data and feed validation, please see our OpenAPI schema description in the OMF [mds-openapi](https://github.com/openmobilityfoundation/mds-openapi) repository. Browsable interactive documentation is also linked to in that repository.
+
+Starting with MDS 2.0, [schema](/schema) validation of endpoints is done using OpenAPI, instead of the former JSON Schema.
```
Comment (thekaveman) [inline README.md:131]:
```suggestion
Starting with MDS 2.0, OpenAPI documents will describe MDS endpoints and allow for [schema](/schema) validation, expanding on the JSON Schemas formerly housed in this repository.
```

Comment (thekaveman) [reply README.md:131]: Just clarifying - all the JSON Schema stuff is still here, it's now just wrapped in OpenAPI documents.

File: schema/README.md [hunk 2]
```diff
@@ -1,34 +1,5 @@
 # MDS Schema Definitions
 
-This directory contains the templates and code that _generate_ the official JSON schemas for the Mobility Data Specification. However, the official schema documents live inside the `provider`, `agency` or appropriate folder.
+Starting with MDS 2.0, schema validation of endpoints is done using [OpenAPI](https://www.openapis.org/) (instead of the former [JSON Schema](https://json-schema.org/)).
```
Comment (thekaveman) [inline schema/README.md:3]:
```suggestion
Starting with MDS 2.0, [OpenAPI](https://www.openapis.org/) documents describe MDS endpoints and allow for schema validation, expanding on the prior JSON Schemas work.
```

Comment (thekaveman) [reply schema/README.md:3]: Another clarification

**Comment by schnuerle** [comment]
Merging this now so finalization work can focus on the touch-ups in the `release-2.0.0` branch to avoid overlapping conflicts with the `feature-openapi` branch. This PR looks great for a first pass at moving to OpenAPI for MDS validation.

**Pull request merged**